pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/assembly/DraftVersionUpload.java

```diff
@@ -0,0 +1,112 @@
+package com.redhat.pantheon.servlet.assembly;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+import com.redhat.pantheon.conf.GlobalConfig;
+import com.redhat.pantheon.model.api.SlingModels;
+import com.redhat.pantheon.model.assembly.Assembly;
+import com.redhat.pantheon.model.assembly.AssemblyLocale;
+import com.redhat.pantheon.model.module.HashableFileResource;
+import com.redhat.pantheon.servlet.ServletUtils;
+import com.redhat.pantheon.servlet.module.ModuleVersionUpload;
+import org.apache.commons.lang3.LocaleUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.servlets.post.AbstractPostOperation;
+import org.apache.sling.servlets.post.Modification;
+import org.apache.sling.servlets.post.PostOperation;
+import org.apache.sling.servlets.post.PostResponse;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import javax.servlet.http.HttpServletResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Locale;
+
+@Component(
+        service = PostOperation.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet POST operation which accepts module uploads and versions them appropriately",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team",
+                PostOperation.PROP_OPERATION_NAME + "=pant:newDraftVersion"
```

Comment:
+1

Reply:
i was trying to make a generic operation that would handle both module and assembly upload. we just ask uploader to pass in a contentType, we can get away with using one upload servlet  .

Reply:
I see. Thing is if we are going to duplicate the classes, we might as well duplicate this one also, otherwise it will just lead to a big if-else with the two different paths (which I guess is another route we could take).

Reply:
I would rather see separate servlets. My gut tells me that as the code evolves, the servlets will look sufficiently different to warrant it. I might be wrong, but we could revisit the decision once the code matures a bit.

Reply:
my initial intuition as well

Reply:
renamed.